torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -727,22 +727,32 @@ Tensor prod_backward(
   if (input.dim() == 0) {
     return grad;
   }
+  if (input.numel() == 0) {
+    return input;
+  }
```

Comment:
1. This is required because the safe path doesn't actually handle it correctly (which we always hit with CCT).

Error:
```
RuntimeError: start (0) + length (-1) exceeds dimension size (0)
```

which fails due:
https://github.com/pytorch/pytorch/blob/6fd8e28a993c578cbdc1e78d71fc2ef71682b165/torch/csrc/autograd/FunctionsManual.cpp#L667-L669

2. It does pass the double backward tests. But I agree that it seems weird. Do you think `grad.view_as(input)` will be robust.

Reply:
Hmm let me think about this a little